File: 1301-1400/1399-Count Largest Group/1399-Count Largest Group.py
import collections
class Solution:
    def countLargestGroup(self, n: int) -> int:
        table = collections.Counter()
        for i in range(1, n + 1):
            sd = 0
            while i:
                sd += i % 10
                i //= 10
            table[sd] += 1
        largestSize = max(table.values())
        return sum(v == largestSize for v in table.values())

File: 1301-1400/1399-Count Largest Group/1399-Count Largest Group.cpp
class Solution {
public:
    int countLargestGroup(int n) {
        unordered_map<int, int> table;
        for (int i = 1; i <= n; ++i) {
            int sum = 0, temp = i;
            while (temp) {
                sum += temp % 10;
                temp /= 10;
            }
            ++table[sum];
        }
        int maxCount = 0, largestSize = 0;
        for (auto p : table) {
            if (p.second > largestSize) {
                largestSize = p.second;
                maxCount = 1;
            }
            else if (p.second == largestSize) {
                ++maxCount;
            }
        }
        return maxCount;
    }
};
